libs/data-mapper/src/lib/utils/DataMap.Utils.ts

```diff
@@ -175,12 +175,14 @@ export const splitKeyIntoChildren = (sourceKey: string): string[] => {
   return results;
 };
 
+export const getSourceLoopKey = (targetKey: string): string => {
+  const forArgs = targetKey.substring(targetKey.lastIndexOf('$for(') + 5, targetKey.lastIndexOf(')'));
```

Comment:
$for -> mapNodeParams.for

`targetKey.lastIndexOf(mapNodeParams.for) + mapNodeParams.for.length + 1`?